File: src/main/java/com/tonight/spark/service/impl/MapServiceImpl.java
package com.tonight.spark.service.impl;

import com.tonight.spark.domain.Map;
import com.tonight.spark.dto.MapInfo;
import com.tonight.spark.repository.MapRepository;
import com.tonight.spark.service.MapService;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.Objects;

@Service
@RequiredArgsConstructor
public class MapServiceImpl implements MapService {

    private final MapRepository repository;

    public String saveUrlInfo(MapInfo urlCollect) {

        String mapName = urlCollect.getMapName();
        String[] split = urlCollect.getMapUrl().split("play/");
        String mapHash = split[1];

        Map findMap = repository.findMapByMapHash(mapHash);
        if (Objects.isNull(findMap)) {
            Map map = Map.builder()
                    .mapHash(mapHash)
                    .mapName(mapName)
                    .build();
            repository.save(map);
        }
        return mapHash;
    }

}

File: src/main/java/com/tonight/spark/service/impl/kotlin/MapServiceImpl.kt
package com.tonight.spark.service.impl.kotlin

import com.tonight.spark.domain.kotlin.ZepMap
import com.tonight.spark.dto.kotlin.MapInfo
import com.tonight.spark.repository.kotlin.ZepMapRepository
import com.tonight.spark.service.kotlin.MapService
import org.springframework.stereotype.Service

@Service
class MapServiceImpl(private val repository: ZepMapRepository) : MapService {

    private val splitter = "/play"

    override fun saveUrlInfo(mapInfo: MapInfo): String {
        val mapName: String = mapInfo.mapName;
        val mapHash: String = mapInfo.mapUrl.split(splitter)[1]
        saveMapHashIfNotExist(mapName, mapHash)
        return mapHash
    }

    private fun saveMapHashIfNotExist(mapName: String, mapHash: String) {
        repository.findByMapHash(mapHash)
                .orElseGet { repository.save(ZepMap(mapName = mapName, mapHash = mapHash)) }
    }
}